app/components/search_bar/search_bar.android.js

```diff
@@ -0,0 +1,178 @@
+import React, {PropTypes, PureComponent} from 'react';
+import {
+    Keyboard,
+    TextInput,
+    StyleSheet,
+    View,
+    TouchableOpacity
+} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+
+const styles = StyleSheet.create({
+    searchBar: {
+        flexDirection: 'row',
+        alignItems: 'center',
+        backgroundColor: 'transparent'
+    },
+    searchBarInput: {
+        flex: 1,
+        fontWeight: 'normal',
+        backgroundColor: 'transparent'
+    }
+});
+
+export default class SearchBarAndroid extends PureComponent {
+    static propTypes = {
+        height: PropTypes.number.isRequired,
+        fontSize: PropTypes.number.isRequired,
+        text: PropTypes.string,
+        placeholder: PropTypes.string,
+        showCancelButton: PropTypes.bool,
+        textFieldBackgroundColor: PropTypes.string,
+        placeholderTextColor: PropTypes.string,
+        textColor: PropTypes.string,
+        onChange: PropTypes.func,
+        onChangeText: PropTypes.func,
+        onFocus: PropTypes.func,
+        onBlur: PropTypes.func,
+        onSearchButtonPress: PropTypes.func,
+        onCancelButtonPress: PropTypes.func
+    };
+
+    static defaultProps = {
+        placeholder: 'Search',
+        showCancelButton: true,
+        placeholderTextColor: '#bdbdbd',
+        textColor: '#212121',
+        onSearchButtonPress: () => true,
+        onCancelButtonPress: () => true,
+        onChangeText: () => true,
+        onFocus: () => true,
+        onBlur: () => true
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isOnFocus: false,
+            value: this.props.text
+        };
+        this.onFocus = this.onFocus.bind(this);
+        this.onBlur = this.onBlur.bind(this);
+        this.onCancelButtonPress = this.onCancelButtonPress.bind(this);
+        this.onSearchButtonPress = this.onSearchButtonPress.bind(this);
+        this.onChangeText = this.onChangeText.bind(this);
+    }
+
+    onSearchButtonPress() {
+        const onSearchButtonPress = this.props.onSearchButtonPress;
+        if (this.state.value) {
+            onSearchButtonPress(this.state.value);
+        }
+    }
+
+    onCancelButtonPress() {
+        const onCancelButtonPress = this.props.onCancelButtonPress;
+        this.setState({
+            isOnFocus: false,
+            value: ''
+        });
+
+        onCancelButtonPress();
+        this.dismissKeyboard();
+    }
+
+    onChangeText(value) {
+        const onChangeText = this.props.onChangeText;
+        this.setState({value});
+        onChangeText(value);
+    }
+
+    onFocus() {
+        const onFocus = this.props.onFocus;
+        this.setState({isOnFocus: true});
+        onFocus();
+    }
+
+    onBlur() {
+        const onBlur = this.props.onBlur;
+        this.setState({isOnFocus: false});
+        onBlur();
+        this.dismissKeyboard();
+    }
+
+    dismissKeyboard() {
```

Comment:
Minor nitpick, looks like this is only wrapping ```Keyboard.dismiss()```. Can we have ```onBlur``` and the ```onCancelButtonPress``` just call ```Keyboard.dismiss()``` directly since this method isn't doing anything other than calling ```Keyboard.dismiss()```? Or I guess we could leave it if there are plans to do other clean ups/actions when dismissing the keyboard.